streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamKStreamJoin.java

```diff
@@ -62,23 +61,25 @@ public void init(ProcessorContext context) {
 
 
         @Override
-        public void process(K key, V1 value) {
-            if (key == null)
+        public void process(final K key, final V1 value) {
+            if (key == null || value == null) {
```

Comment:
maybe add a comment explaining why we are excluding `<key: null>` for stream stream join, pros and cons compared with treating each `<key: null>` as a different message?